include/ebpf_api.h

```diff
@@ -417,6 +417,22 @@ extern "C"
         size_t attach_parameters_size,
         _Outptr_ struct bpf_link** link) EBPF_NO_EXCEPT;
 
+    /**
+     * @brief Attach an eBPF program by program file descriptor and return
+     * the link as a file descriptor.
+     *
+     * @see ebpf_program_attach_by_fd
+     *
+     * @retval EBPF_SUCCESS The operation was successful.
+     */
+    _Must_inspect_result_ ebpf_result_t
+    ebpf_program_attach_fds(
```

Comment:
nit: just for understanding, as an alternative, is it sufficient to use `ebpf_program_attach_by_fd()` followed by `bpf_link__fd()` instead of this new API?

Reply:
Unfortunately this doesn't work because there is no way to deallocate a `*bpf_link` without closing the fd / handle. In my first iteration I added a function which frees the link and returns the fd. It seemed to me that this approach only works well for links, because they are very simple objects and cheap to allocate / deallocate. I'll need to add a similar API for native image loads, and there the "unwrapping" approach doesn't work well.

Reply:
What about adding ebpf_get_handle_from_fd? (And perhaps ebpf_get_fd_from_handle if needed).  The former might allow using some additional Windows APIs that work with handles.

Reply:
Conclusion from the office hours: we keep the current approach of using fd_t and will add the handle_from_fd.